state_synchronizer/src/peer_manager.rs

```diff
@@ -76,6 +79,10 @@ impl PeerManager {
             self.peers
                 .insert(peer_id, PeerInfo::new(true, false, MAX_SCORE));
         }
+
+        if self.weighted_index.is_none() {
```

Comment:
if it's a new peer, but not yet included in index, index should be rebuilded

also disable_peer might effect score